packages/story-editor/src/components/floatingMenu/elements/textAlign.js

```diff
@@ -17,21 +17,165 @@
 /**
  * External dependencies
  */
-import { Icons } from '@googleforcreators/design-system';
+import { useState, useRef, useEffect } from '@googleforcreators/react';
+import {
+  Icons,
+  ContextMenu,
+  ContextMenuComponents,
+  TOOLTIP_PLACEMENT,
+  useKeyDownEffect,
+} from '@googleforcreators/design-system';
 import { __ } from '@googleforcreators/i18n';
+import styled from 'styled-components';
+import { trackEvent } from '@googleforcreators/tracking';
 
 /**
  * Internal dependencies
  */
-import { IconButton } from './shared';
+import { useStory } from '../../../app';
+import { IconButton, useProperties } from './shared';
+
+const OFFSET_X = -8;
+const OFFSET_Y = 3;
+
+const SubMenuContainer = styled.div`
+  position: absolute;
+  top: calc(var(--height) + ${OFFSET_Y}px);
+  z-index: 9999;
+`;
+
+const StyledMenuButton = styled(ContextMenuComponents.MenuButton)`
+  flex: 0 0 60px;
+  height: 36px;
+  border: 1px solid ${({ theme }) => theme.colors.border.defaultNormal};
+  padding: 2px 0;
+  border-radius: 4px;
+`;
+
+const StyledMenuIcon = styled(ContextMenuComponents.MenuIcon)`
+  width: auto;
+  display: flex;
+  overflow: hidden;
+
+  svg {
+    width: 32px;
+  }
+
+  svg:last-child {
+    margin-left: -8px;
+  }
+`;
+
+const ALIGNMENTS = [
+  {
+    value: 'left',
+    icon: Icons.AlignTextLeft,
+    label: __('Align text left', 'web-stories'),
+  },
+  {
+    value: 'center',
+    icon: Icons.AlignTextCenter,
+    label: __('Align text center', 'web-stories'),
+  },
+  {
+    value: 'right',
+    icon: Icons.AlignTextRight,
+    label: __('Align text right', 'web-stories'),
+  },
+  {
+    value: 'justify',
+    icon: Icons.AlignTextJustified,
+    label: __('Align text justified', 'web-stories'),
+  },
+];
 
 function TextAlign() {
+  const { textAlign = 'left' } = useProperties(['textAlign']);
+  const updateSelectedElements = useStory(
+    (state) => state.actions.updateSelectedElements
+  );
+
+  const [isMenuOpen, setMenuOpen] = useState(false);
+  const subMenuRef = useRef();
+  const buttonRef = useRef();
+  const [offsetLeft, setOffsetLeft] = useState(0);
+
+  // Record left position of this button in the parent design menu
+  useEffect(() => setOffsetLeft(buttonRef.current.offsetLeft + OFFSET_X), []);
+
+  // When menu has just opened, focus the current button in submenu
+  const currentIconMounted = (node) => {
+    if (node) {
+      /* eslint-disable-next-line @wordpress/react-no-unsafe-timeout
+         --------
+         We have to wait a frame for the parent to have rendered
+      */
+      setTimeout(() => node.focus());
+    }
+  };
+
+  const { icon: CurrentIcon, label: currentLabel } =
+    ALIGNMENTS.find(({ value }) => value === textAlign) || ALIGNMENTS[0];
+
+  const handleTextAlign = (value) => () => {
+    trackEvent('floating_menu', {
+      name: `set_text_alignment_${value}`,
```

Comment:
@swissspidy,@miina: would this be the proper way to do it or should the actual alignment value be (another) property on the object?

Reply:
Sgtm as is if @swissspidy is ok with it -- in some other places we're also setting the name dynamically like this.

Reply:
I think it's fine.

We can do a more holistic analysis for floating menu bar telemetry later, to ensure coherent naming.